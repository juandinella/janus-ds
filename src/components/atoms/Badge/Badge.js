import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './Badge.styles'

/**
 * Badge
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledBadge = styled.div.withConfig({shouldForwardProp})`${styles}`

const Badge = ({ id }) => {
  return <StyledBadge id={id}/>
}

Badge.displayName = 'Badge'

Badge.propTypes = {
  /** Component id */
  id: PropTypes.string,
}

Badge.defaultProps = {}

export default Badge
