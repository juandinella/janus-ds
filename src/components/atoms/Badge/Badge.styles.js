import { css } from 'styled-components'

export default css`
  --size: 10px;
  width: var(--size);
  height: var(--size);
  min-width: var(--size);
  min-height: var(--size);
  background-color: var(--bg-primary);
  border-radius: 100%;
`
