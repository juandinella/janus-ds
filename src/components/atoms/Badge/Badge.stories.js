import Badge from './Badge'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(Badge)

export default {
  title: 'Atoms/Badge',
  component: Badge,
  args: {},
  argTypes: {},
}

export const Default = Template.bind({})
