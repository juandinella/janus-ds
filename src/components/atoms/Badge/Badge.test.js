import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import Badge from './Badge'

initStoryshots({
  storyKindRegex: /^Atoms\/Badge$/,
})

const componentProps = {
  id: 'unique_id',
}
afterEach(cleanup)

describe('<Badge />', () => {
  test('should render component correctly', () => {
    render(<Badge {...componentProps} />)
    const bdg = screen.getByTestId(componentProps.id)
    expect(bdg).toBeInTheDocument()
  })
})
