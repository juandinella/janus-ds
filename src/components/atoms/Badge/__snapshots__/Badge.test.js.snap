// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Badge Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-2gfkx2-0 kuaYtC"
  />
</div>
`;
